lib/Services/RoutingService.php

```diff
@@ -55,13 +55,14 @@ public static function call(string $controllerClassname, bool $authenticated = t
         if ($response instanceof Template) {
             $response->data['messages'] = $container->get(SessionMessagesService::class)->getMessages();
 
-            $response->show();
+            return $response->show();
```

Comment:
Template::show() never returns a value, so this is wrong..

Reply:
Fixed in https://github.com/rediris-es/simplesamlphp-module-oidc/pull/36/commits/f799ec1327748d02ee8a4914e19fffdeea4f9b2c

Reply:
I don't understand why you want to add the return.. It's of no use..
Anyway, it's up to the maintainer to accept this PR.. I don't have write-access

Reply:
If there is no return then the function throws an exception. 
https://github.com/rediris-es/simplesamlphp-module-oidc/blob/master/lib/Services/RoutingService.php#L67

Reply:
Ah, I think I understand.. but it's not this one, but the second case!

Reply:
Template::show never returns, so it's safe to remove the return here

Reply:
Without the return after ```show()``` I get this error navigating to ```module.php/oidc/clients/```
```
[Wed Mar 25 17:59:13.778239 2020] [php7:notice] [pid 14] [client 172.24.0.2:36660] simplesamlphp ERR [e069591f73] SimpleSAML\\Error\\Error: UNHANDLEDEXCEPTION, referer: http://siu:8286/siu/idp/module.php/core/frontpage_federation.php
[Wed Mar 25 17:59:13.778481 2020] [php7:notice] [pid 14] [client 172.24.0.2:36660] simplesamlphp ERR [e069591f73] Backtrace:, referer: http://siu:8286/siu/idp/module.php/core/frontpage_federation.php
[Wed Mar 25 17:59:13.778564 2020] [php7:notice] [pid 14] [client 172.24.0.2:36660] simplesamlphp ERR [e069591f73] 1 /usr/local/app/idp/vendor/simplesamlphp/simplesamlphp/www/_include.php:17 (SimpleSAML_exception_handler), referer: http://siu:8286/siu/idp/module.php/core/frontpage_federation.php
[Wed Mar 25 17:59:13.778626 2020] [php7:notice] [pid 14] [client 172.24.0.2:36660] simplesamlphp ERR [e069591f73] 0 [builtin] (N/A), referer: http://siu:8286/siu/idp/module.php/core/frontpage_federation.php
[Wed Mar 25 17:59:13.778692 2020] [php7:notice] [pid 14] [client 172.24.0.2:36660] simplesamlphp ERR [e069591f73] Caused by: SimpleSAML\\Error\\Exception: Response type not supported: SimpleSAML\\XHTML\\Template, referer: http://siu:8286/siu/idp/module.php/core/frontpage_federation.php
[Wed Mar 25 17:59:13.778746 2020] [php7:notice] [pid 14] [client 172.24.0.2:36660] simplesamlphp ERR [e069591f73] Backtrace:, referer: http://siu:8286/siu/idp/module.php/core/frontpage_federation.php
[Wed Mar 25 17:59:13.778798 2020] [php7:notice] [pid 14] [client 172.24.0.2:36660] simplesamlphp ERR [e069591f73] 3 /usr/local/app/idp/vendor/simplesamlphp/simplesamlphp/modules/oidc/lib/Services/RoutingService.php:67 (SimpleSAML\\Modules\\OpenIDConnect\\Services\\RoutingService::call), referer: http://siu:8286/siu/idp/module.php/core/frontpage_federation.php
[Wed Mar 25 17:59:13.778907 2020] [php7:notice] [pid 14] [client 172.24.0.2:36660] simplesamlphp ERR [e069591f73] 2 /usr/local/app/idp/vendor/simplesamlphp/simplesamlphp/modules/oidc/www/clients/index.php:18 (require), referer: http://siu:8286/siu/idp/module.php/core/frontpage_federation.php
[Wed Mar 25 17:59:13.778967 2020] [php7:notice] [pid 14] [client 172.24.0.2:36660] simplesamlphp ERR [e069591f73] 1 /usr/local/app/idp/vendor/simplesamlphp/simplesamlphp/lib/SimpleSAML/Module.php:254 (SimpleSAML\\Module::process), referer: http://siu:8286/siu/idp/module.php/core/frontpage_federation.php
[Wed Mar 25 17:59:13.779021 2020] [php7:notice] [pid 14] [client 172.24.0.2:36660] simplesamlphp ERR [e069591f73] 0 /usr/local/app/idp/vendor/simplesamlphp/simplesamlphp/www/module.php:10 (N/A), referer: http://siu:8286/siu/idp/module.php/core/frontpage_federation.php
[Wed Mar 25 17:59:13.779133 2020] [php7:notice] [pid 14] [client 172.24.0.2:36660] simplesamlphp ERR [e069591f73] Error report with id e81c96a1 generated., referer: http://siu:8286/siu/idp/module.php/core/frontpage_federation.php
```
I'm testing in PHP 7.3, maybe it's related?

Reply:
I think the issue is (type) confusion..  We're not actually dealing with a `SimpleSAML\XHTML\Template`, but with a `ReflectionClass`...  I believe @sgomez was going to rewrite this part of the module to use Symfony Response-objects instead, so for now I can live with the added returns..